src/joint.dia.paper.js

```diff
@@ -620,6 +620,17 @@ joint.dia.Paper = Backbone.View.extend({
 
     snapToGrid: function(p) {
 
+        // Synthesized events such as in unit tests won't have clientX/clientY
```

Comment:
Are you able to add some dummy `clientX` and `clientY` to synthesized events in your unit test?